addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/protocol/utils/VolumeConverter.java

```diff
@@ -74,13 +81,21 @@ public static double convertFromIpControlVolumeToPercent(String ipControlVolume)
      * @param ipControlVolume
      * @return
      */
-    private static String formatIpControlVolume(double ipControlVolume) {
-        String result = IP_CONTROL_VOLUME_DEFAULT_VALUE;
+    private static String formatIpControlVolume(double ipControlVolume, int zone) {
+        validateZone(zone - 1);
+        DecimalFormat FORMATTER = new DecimalFormat(IP_CONTROL_VOLUME_FORMAT[zone - 1]);
+        String result = IP_CONTROL_VOLUME_DEFAULT_VALUE[zone - 1];
         // DecimalFormat is not ThreadSafe
         synchronized (FORMATTER) {
             result = FORMATTER.format(Math.round(ipControlVolume));
         }
         return result;
     }
 
+    private static void validateZone(int zone) {
+        if (zone < 0 || zone > 3) {
+            throw new IllegalArgumentException("An unexpected zone was received, the value should be in the range 0-3");
```

Comment:
I personally would have left out the repeated zone - 1 in calling it, now the exception is not in line with the actual zone which passed in to the class. What do you think?

Reply:
yes.. i don't like those calls. But i favored doing the check with the same index the array is accessed afterwards with for now.
I am planning on some future PRs on this binding introducing more features and was planning to possibly get rid of the "-1" calls completely then.